### ISSUE TITLE
How to allow undocumented responses?

### ISSUE THREAD

**Issue by pumazi** [Yelp/bravado#329]
## tl;dr;
Is there a way to allow "undocumented" responses? I use "undocumented" because this is how the [Swagger Editor](http://editor.swagger.io/#) terms it.

## long description of the problem

Originally, I was trying to use the [example in "Let's try a POST call" of the documentation](https://bravado.readthedocs.io/en/latest/quickstart.html#lets-try-a-post-call). This didn't seem to work due to a validation problem. So things aren't working as documented, but that's a different issue. The following is my attempt to debug why the documented example doesn't work.

In trying to debug this issue I replicated parts of `tests/petstore/pet/addPet_test.py`, which were more successful, but still failed. The following is the response of that experiment.

```pytb
>>> result = petstore.pet.addPet(body=fido).result()
Traceback (most recent call last):
  File "$master_checkout/bravado/bravado/http_future.py", line 153, in unmarshal_response
    operation)
  File "$venv/lib/python3.5/site-packages/bravado_core/response.py", line 96, in unmarshal_response
    response_spec = get_response_spec(response.status_code, op)
  File "$venv/lib/python3.5/site-packages/bravado_core/response.py", line 151, in get_response_spec
    "status_code or use a `default` response.".format(status_code, op))
bravado_core.exception.MatchingResponseNotFound: Response specification matching http status_code 200 not found for operation Operation(addPet). Either add a response specification for the status_code or use a `default` response.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "$master_checkout/bravado/bravado/http_future.py", line 56, in wrapper
    return func(self, *args, **kwargs)
  File "$master_checkout/bravado/bravado/http_future.py", line 115, in result
    self.response_callbacks)
  File "$master_checkout/bravado/bravado/http_future.py", line 162, in unmarshal_response
    sys.exc_info()[2])
  File "$venv/lib/python3.5/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "$master_checkout/bravado/bravado/http_future.py", line 153, in unmarshal_response
    operation)
  File "$venv/lib/python3.5/site-packages/bravado_core/response.py", line 96, in unmarshal_response
    response_spec = get_response_spec(response.status_code, op)
  File "$venv/lib/python3.5/site-packages/bravado_core/response.py", line 151, in get_response_spec
    "status_code or use a `default` response.".format(status_code, op))
bravado.exception.HTTPError: 200 OK: Response specification matching http status_code 200 not found for operation Operation(addPet). Either add a response specification for the status_code or use a `default` response.
```

Initializing the client with `SwaggerClient(..., config={'validate_responses': False})` produces the same results.

A valid response with the same data is produced with the Swagger Editor, so the api works. What I'd like to know is how I can **allow undocumented/undefined responses**.

**Comment by pumazi** [comment]
This may or may not be slightly connected to #314 (CC @macisamuele).

**Comment by sjaensch** [comment]
@pumazi this is not supported currently. Are you not able to either add a default response or a response with the specified code? You don't need to define the response schema - all you need is a proper entry in the `responses` section. Check out [the documentation](https://swagger.io/docs/specification/2-0/describing-responses/) for more information and examples.

Note that you don't need to define the schema of the response - neither for specific error codes nor for the default one.
